staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/types.go

```diff
@@ -20,15 +20,20 @@ import (
 	"context"
 	"time"
 
+	genericrequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/util/flowcontrol/debug"
 	"k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/promise"
 )
 
 // request is a temporary container for "requests" with additional
 // tracking fields required for the functionality FQScheduler
 type request struct {
-	qs     *queueSet
-	fsName string
-	ctx    context.Context
+	ctx context.Context
```

Comment:
this looks improper.  The recommendation is to avoid storing context.  If we want request info and user info, let's store exactly those.

Reply:
> this looks improper. The recommendation is to avoid storing context. If we want request info and user info, let's store exactly those.

hm, improper, but pre-existing.  Maybe it's for a different purpose somewhere